platforms/azure/modules/lbaas/lbaas.tf

```diff
@@ -47,6 +47,13 @@ resource "azurerm_public_ip" "lb" {
   public_ip_address_allocation = "Static"
 }
 
+data "azurerm_public_ip" "lb" {
```

Comment:
Could explain, what this is for. You know Azure for sure better then me. Is this special for the azurerm. Why do you need the data if the ip is generated by a regular resource?

Reply:
in azure, public ip's are allocated and set after the resource is created. So Terraform won't have the knowledge of this public ip at deploy time. This makes it impossible to deploy with a public ip. To solve this, there is this public_ip resource. This ensures the ip is allocated before the wanted resource is deployed.

Reply:
Yepp I joust thought somethinglike this. The Azurerm provider is not really convincing. There are so many problems.

We could not solve the major problem: handling of attached disks for the vm resource. If used incrementally (not only for creation) the resource does not handle dynamically attached resources (by the kubernetes controllers to persistent volumes). As a result it is NOT possible to
do day-2 operations (kubify was designed for) on Azure.

We tried to address this problem at kubecon, but we couldn't find a valid contact.